### DISCUSSION TITLE
UITableView with insetGrouped style

### DISCUSSION THREAD

**Discussion by nizarjak** [pointfreeco/swift-snapshot-testing#502] [Q&A]
Noticed an issue while snapshotting UITableView with insetGrouped style. It layouts correctly during run but on snapshots it has artifacts between first & second and second last & last rows in a section.

<img width="414" alt="testExample 1" src="https://user-images.githubusercontent.com/6884147/82201411-9cec4100-9900-11ea-9541-2f9880e7e0c2.png">

Does anyone have any idea what happened? Tried delaying layout before taking snapshot, but that was not the case. Snapshot are consistent throughout multiple runs.

I've created a simple project with as little code as possible to replicate the issue. Snapshot taken from simulator iPhone 11 (13.4.1) using Xcode 11.4.1.

[GroupedTableViewVC.zip](https://github.com/pointfreeco/swift-snapshot-testing/files/4643494/GroupedTableViewVC.zip)

**Comment by piercifani** [comment]
Yep, happening here as well... 😢

**Comment by jaetzold** [comment]
Same happens when using https://github.com/uber/ios-snapshot-test-case btw.

**Comment by jaetzold** [comment]
Probably related to https://github.com/pointfreeco/swift-snapshot-testing/issues/358

**Comment by lukeredpath** [comment]
I had this problem on my current project - the workaround was to set `drawHierarchyInKeyWindow: true` which causes the library to render the view using `drawHierarchy(in:afterScreenUpdates:)` instead.

Worth noting that this workaround may introduce other problems - whilst `drawInContext` uses the CPU, `drawHierarchy(in:afterScreenUpdates:)` uses the GPU and may result in inconsistent results on different machines, particularly CI as I'm currently struggling with (see #419).

**Comment by stephencelis** [comment]
I'm going to convert this to a discussion since it seems to be a more general bug that affects other frameworks, and @lukeredpath has a workaround. If someone figures out how to apply this fix directly in a strategy, we'd definitely take a fix, but for now going to surface discussion over on the forums so that issues can focus on bugs in the library.
